### PR TITLE
new: keyless auth using github action token

### PR DESCRIPTION
**Pull request by ankurv3rma**
This PR introduces a new boolean input parameter `enable_github_action_token` to enable keyless auth in the github action. `enable_github_action_token` is set to `false` by default. When it is set to true, api key/secret or GCP service account inputs are not required and will be ignored.

**Requisites**
- User must create an authorization rule in the UI
- User must enable id-token: write permissions in their github action

An example run with this parameter can be found here:
https://github.com/ankurv3rma/vulnado-fork/actions/runs/4322573676/jobs/7545148854

### CODE REVIEW

**Comment by ankurv3rma** [comment]
There is a known issue with the `Code Scanner` permission in the authorization rule and the scan fails because of that. But this currently works with the `Admin` permission. We can wait till it is resolved to merge this.
cc @dstiliadis

File: dist/index.js [hunk 1]
```diff
@@ -13637,10 +13640,17 @@ function run() {
...
-            if (GCP_CREDENTIALS_SERVICE_ACCOUNT)
-                options.push(`--gcp-service-account=${GCP_CREDENTIALS_SERVICE_ACCOUNT}`);
+            if (ENABLE_GITHUB_ACTION_TOKEN) {
+                options.push(`--enable-github-action-token=true`);
+            }
+            else {
+                if (API_KEY && API_SECRET) {
+                    options.push(`--api-key=${API_KEY}`, `--api-secret=${API_SECRET}`);
+                }
+                if (GCP_CREDENTIALS_SERVICE_ACCOUNT) {
```
Comment (dstiliadis) [inline dist/index.js]: I think we need a switch type statement here. Note, that if API_KEY and SECRET are there, then we should not push the GCP CREDENTIALS (unless if this checked somewhere else and the two cannot be set at the same time)

File: dist/index.js [hunk 2]
```diff
@@ -13616,7 +13617,9 @@ function run() {
                 core.setFailed("namespace is required and must be passed as an input from the workflow");
                 return;
             }
-            if (!(API_KEY && API_SECRET) && !GCP_CREDENTIALS_SERVICE_ACCOUNT) {
+            if (!ENABLE_GITHUB_ACTION_TOKEN &&
+                !(API_KEY && API_SECRET) &&
+                !GCP_CREDENTIALS_SERVICE_ACCOUNT) {
```
Comment (dstiliadis) [inline dist/index.js:13623]: at least one of them must be provided. The above checks if none of them is provided. But we need a check if at least one of them is provided.

**Comment by dstiliadis** [comment]
> There is a known issue with the `Code Scanner` permission in the authorization rule and the scan fails because of that. But this currently works with the `Admin` permission. We can wait till it is resolved to merge this. cc @dstiliadis

This is addressed separately and not required for this. We can proceed with this one after the comments are resolved.

File: dist/index.js (hunk 1 above)
Comment (ankurv3rma) [reply dist/index.js]: This was not checked anywhere else. Fixed now.

File: dist/index.js (hunk 2 above)
Comment (ankurv3rma) [reply dist/index.js:13623]: The check here is to fail and display an error if none of them is provided. If atleast one of them is provided, the run would proceed

**Review by dstiliadis — APPROVED** [review]

**Pull request merged**